src/connections/Validation.ts

```diff
@@ -0,0 +1,142 @@
+import { isEmpty } from "lodash";
+import { useCallback, useState } from "react";
+
+import { v3Resource } from "@/connections/util/apiConnectionFactory";
+import {
+  getPolygonValidation,
+  GetPolygonValidationPathParams,
+  getSiteValidation
+} from "@/generated/v3/researchService/researchServiceComponents";
+import { ValidationDto } from "@/generated/v3/researchService/researchServiceSchemas";
+import { useConnection } from "@/hooks/useConnection";
+import ApiSlice from "@/store/apiSlice";
+import { loadConnection } from "@/utils/loadConnection";
+
+const hasValidParams = (pathParams: GetPolygonValidationPathParams | undefined): boolean => {
+  const isValid =
+    pathParams != null &&
+    pathParams.polygonUuid != null &&
+    pathParams.polygonUuid !== "" &&
+    pathParams.polygonUuid !== "undefined" &&
+    !isEmpty(pathParams.polygonUuid);
+
+  return isValid;
+};
```

Comment:
There's no reason to assign the value and then do nothing but return it (my editor actually warns about this), so this can be a simple inline function:
```
const hasValidParams = (pathParams: GetPolygonValidationPathParams | undefined): boolean =>
  pathParams != null &&
  pathParams.polygonUuid != null &&
  pathParams.polygonUuid !== "" &&
  pathParams.polygonUuid !== "undefined" &&
  !isEmpty(pathParams.polygonUuid);
```

Reply:
This can also be a lot simpler. The isEmpty term catches a lot of whats above, so this might more simply be:
```
!isEmpty(pathParams?.polygonUuid) && pathParams?.polygonUuid !== "undefined"
```